shell/common/shell_test.cc

```diff
@@ -22,6 +22,22 @@ ShellTest::ShellTest()
 
 ShellTest::~ShellTest() = default;
 
+void ShellTest::SendEnginePlatformMessage(
+    Shell* shell,
+    fml::RefPtr<PlatformMessage> message) {
+  fml::AutoResetWaitableEvent latch;
+  fml::TaskRunner::RunNowOrPostTask(
+      shell->GetTaskRunners().GetPlatformTaskRunner(),
+      [shell, &latch, message = std::move(message)]() {
+        if (!shell->weak_engine_) {
+          return;
```

Comment:
You missed signaling the latch here. Consider instead:

```
if (auto engine = shell->weak_engine_) {
  engine->Handle(...);
}
latch.Signal();
```

Reply:
Ahh right, thanks.  Fixed.